### PR TITLE
switching channel prompts to run flutter upgrade

### LINKED ISSUE #16030: Switching channels should prompt to run flutter upgrade
## Steps to Reproduce
Was on beta channel, and ran the following:

```
$ flutter channel master                                                        
Switching to flutter channel 'master'...
git: Switched to branch 'master'
git: Your branch is behind 'origin/master' by 272 commits, and can be fast-forwarded.
git:   (use "git pull" to update your local branch)
```

The message here is technically accurate (the best kind of accurate), but confusing in that it suggests the next step is to run `git pull`. It would be better if it instead closed with something like the following (assuming no error was returned from git):

```
Successfully switched to flutter channel 'master'.
Now run "flutter upgrade" to ensure that you're on the latest build from this channel.
```

(Or even better, perhaps we should automatically run `flutter upgrade` after switching channels, since the tools are otherwise in an indeterminate state.)

### PR DESCRIPTION
**Pull request by mannprerak2**
## Description

Added print statements to prompt user to run `flutter upgrade` after running `flutter channel foo`

## Related Issues

resolves #16030

## Tests
None... simply added print statements

Here's an output from my system:
```
Switching to flutter channel 'beta'...
git: Switched to branch 'beta'
git: Your branch is up-to-date with 'origin/beta'.
Successfully switched to flutter channel 'beta'.
Now run "flutter upgrade" to ensure that you're on the latest build from this channel.
```
## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] All existing and new tests are passing.
- [ ] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by mannprerak2** [comment]
Can this also close #14979 ?

**Comment by zanderso** [comment]
@jonahwilliams is this consistent with your plans for `flutter upgrade` ?

**Review by jonahwilliams — CHANGES_REQUESTED** [review]
I would adjust the phrasing slightly. Maybe "To ensure you're on the lastest build for this channel, run flutter upgrade"

**Comment by jonahwilliams** [comment]
There are some existing tests in https://github.com/flutter/flutter/blob/master/packages/flutter_tools/test/commands.shard/permeable/upgrade_test.dart

that check if stdout contains the right strings, this addition should have a basic smoke test there

**Comment by mannprerak2** [comment]
> https://github.com/flutter/flutter/blob/master/packages/flutter_tools/test/commands.shard/permeable/upgrade_test.dart

@jonahwilliams but the changes are related to running `flutter channel`, where should I add those tests?

**Comment by jonahwilliams** [comment]
Oh, sorry I wasn't thinking. It would go in the channel_test.dart file

**Comment by mannprerak2** [comment]
@jonahwilliams added a test

File: packages/flutter_tools/test/general.shard/channel_test.dart [hunk 1]
```diff
@@ -253,6 +253,34 @@ void main() {
...
+        <String>['git', 'checkout', 'beta', '--'],
+        workingDirectory: anyNamed('workingDirectory'),
+        environment: anyNamed('environment'),
+      )).thenAnswer((_) => Future<Process>.value(createMockProcess()));
+
+      final ChannelCommand command = ChannelCommand();
+      final CommandRunner<void> runner = createTestCommandRunner(command);
+      await runner.run(<String>['channel', 'beta']);
+
+      expect(testLogger.statusText, contains("To ensure that you're on the latest build from this channel, run 'flutter upgrade'"));
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/channel_test.dart]: Make sure that we output the right branch name too

**Review by jonahwilliams — APPROVED** [review]
LGTM with nit

**Comment by mannprerak2** [comment]
@jonahwilliams I added the complete line `Successfully switched to flutter channel 'beta'.`
Also, If this seems fine, can you add the label for automated merging

**Comment by jonahwilliams** [comment]
Thanks for the contribution!

**Comment by jonahwilliams** [comment]
Looks like this change broke another test:

```
00:25 +880 ~5 -1: test/general.shard/channel_test.dart: channel removes version stamp file when switching channels [E]                                                                                 
  Expected: <1>
    Actual: <2>
```

@mannprerak2 can you take a look?

**Comment by mannprerak2** [comment]
Fixed.. I didn't know calls to verify were neccesary
```dart
verify(mockProcessManager.start(
        <String>['git', 'fetch'],
        workingDirectory: anyNamed('workingDirectory'),
        environment: anyNamed('environment'),
      )).called(1);
```

**Comment by mannprerak2** [comment]
@jonahwilliams can you look into why tool_tests-commands-macos test is failing?
If it's some other network or device error shouldn't this be set to automatic re-run?

**Pull request merged**